lib/test-transformer.js

```diff
@@ -0,0 +1,64 @@
+var cachingTransform = require('caching-transform');
+var fs = require('fs');
+var path = require('path');
+var crypto = require('crypto');
+var cacheDir = path.join(module.paths[1], '.cache', 'ava');
+var filenameToHash = {};
+
+function factory(cacheDir) {
+	var createEspowerPlugin = require('babel-plugin-espower/create');
+	var babel = require('babel-core');
+	var enhanceAssert = require('./enhance-assert');
+	var convertSourceMap = require('convert-source-map');
+
+	function buildOptions(filename, code) {
+		// initialize power-assert
+		var powerAssert = createEspowerPlugin(babel, {
+			patterns: enhanceAssert.PATTERNS
+		});
+
+		var sourceMap = convertSourceMap.fromSource(code) || convertSourceMap.fromMapFileSource(code, path.dirname(filename));
+
+		return {
+			presets: [require('babel-preset-stage-2'), require('babel-preset-es2015')],
+			plugins: [powerAssert, require('babel-plugin-transform-runtime')],
+			filename: filename,
+			sourceMaps: true,
+			ast: false,
+			inputSourceMap: sourceMap && sourceMap.toObject()
+		};
+	}
+
+	return function (code, filename, hash) {
+		var options = buildOptions(filename, code);
+		var result = babel.transform(code, options);
+		var mapFile = path.join(cacheDir, hash + '.map');
+		fs.writeFileSync(mapFile, JSON.stringify(result.map));
+		return result.code;
+	};
+}
+
+var transform = cachingTransform({
+	factory: factory,
+	cacheDir: cacheDir,
+	ext: '.js',
+	hash: function (code, filename, salt) {
+		var hash = crypto
+			.createHash('md5')
+			.update(code, 'utf8')
+			.update(filename || 'unknown file', 'utf8')
+			.update(salt || '', 'utf8')
```

Comment:
Just curious, is there any difference in using `.update()` multiple times rather than just concatenating the strings together before passing them to `.update()`?